### PR TITLE
gtest: remove lib/gtest and add as library

### PR DESCRIPTION
**Pull request by lrusak**
This removes the lib/gtest folder and adds gtest as a separate library that can be disabled.

This allows us to bump gtest easier (not that it seems to be released often). It also allows anyone to disable building the tests as they are currently built if the target matches the host. This should be helpful for distros that don't need to run the tests.

This adds a few new switches:
 - `ENABLE_TESTING` = used to enable/disable building the kodi test suit (enabled by default)
 - `ENABLE_INTERNAL_GTEST ` = used to force building the internal gtest library (incase the os doesn't provide the correct version)
 - `GTEST_URL` = use to specify the path to the gtest archive (useful for offline building)

adding gtest to the depends isn't needed as the tests are only able to run on the target anyways (not on the host).

### CODE REVIEW

**Comment by garbear** [comment]
loving those line counts

**Comment by lrusak** [comment]
jenkins build this please

File: cmake/modules/FindGtest.cmake [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  else()
+    set(GTEST_URL http://mirrors.kodi.tv/build-deps/sources/googletest-${GTEST_VERSION}.tar.gz)
+  endif()
+
+  if(VERBOSE)
+    message(STATUS "GTEST_URL: ${GTEST_URL}")
+  endif()
+
+  if(CORE_SYSTEM_NAME STREQUAL windows)
+    set(GOOGLETEST_CMAKE_ARGS -DCMAKE_CXX_FLAGS=/D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING -DBUILD_SHARED_LIBS=ON)
```
Comment (garbear) [inline cmake/modules/FindGtest.cmake]: Is `/D` intentional?

File: CMakeLists.txt [hunk 2]
```diff
@@ -352,16 +351,18 @@ else()
...
-unset(_TEST_LIBRARIES)
-add_dependencies(${APP_NAME_LC}-test ${APP_NAME_LC}-libraries export-files)
-
-# Enable unit-test related targets
-if(CORE_HOST_IS_TARGET)
+if(CORE_HOST_IS_TARGET AND ENABLE_TESTING)
+  find_package(Gtest)
+  copy_files_from_filelist_to_buildtree(${CMAKE_SOURCE_DIR}/cmake/installdata/test-reference-data.txt NO_INSTALL)
+  add_executable(${APP_NAME_LC}-test EXCLUDE_FROM_ALL ${CMAKE_SOURCE_DIR}/xbmc/test/xbmc-test.cpp ${test_sources})
+  set_target_properties( PROPERTIES ENABLE_EXPORTS ON)
```
Comment (garbear) [inline CMakeLists.txt]: extraneous space. also this block could use some newlines whitespace

**Review by DaveTBlake — APPROVED** [review]

File: cmake/modules/FindGtest.cmake (hunk 1 above)
Comment (lrusak) [reply cmake/modules/FindGtest.cmake]: it was but it's not needed anymore I guess

File: CMakeLists.txt (hunk 2 above)
Comment (lrusak) [reply CMakeLists.txt]: done, thanks!

**Pull request merged**